workitem/workitem_repository.go

```diff
@@ -690,48 +693,173 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
 	}
+	wiMap := map[string]WICountsPerIteration{}
+	for _, r := range res {
+		wiMap[r.IterationID] = WICountsPerIteration{
+			IterationID: r.IterationID,
+			Total:       r.Total,
+			Closed:      r.Closed,
+		}
+	}
+	// put 0 count for iterations which are not in wiMap
+	// ToDo: Update count query to include non matching rows with 0 values
+	// Following operation can be skipped once above is done
+	for _, i := range allIterations {
+		if _, exists := wiMap[i.String()]; exists == false {
+			wiMap[i.String()] = WICountsPerIteration{
+				IterationID: i.String(),
+				Total:       0,
+				Closed:      0,
+			}
+		}
+	}
+
+	type IterationHavingChildrenID struct {
+		Children    string `gorm:"column:children"`
+		IterationID string `gorm:"column:iterationid"`
+	}
+	var itrChildren []IterationHavingChildrenID
+	queryIterationWithChildren := fmt.Sprintf(`
+	WITH PathResolver AS
+	(SELECT CASE
+				WHEN path = '' THEN replace(id::text, '-', '_')::ltree
+				ELSE concat(path::text, '.', REPLACE(id::text, '-', '_'))::ltree
+			END AS pathself,
+			id
+	FROM %s)
+	SELECT array_agg(iterations.id)::text AS children,
+		PathResolver.id::text AS iterationid
+	FROM %s,
+		PathResolver
+	WHERE path <@ PathResolver.pathself
+	AND space_id = ?
+	GROUP BY (PathResolver.pathself,
+		PathResolver.id)`,
+		iterationTableName,
+		iterationTableName)
+	db = r.db.Raw(queryIterationWithChildren, spaceID.String())
+	db.Scan(&itrChildren)
+	if db.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"space_id": spaceID.String(),
+			"err":      db.Error,
+		}, "unable to fetch children for every iteration in a space")
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	childMap := map[string][]string{}
+	for _, r := range itrChildren {
+		// Following can be done by implementing Valuer interface for type IterationHavingChildrenID
+		r.Children = strings.TrimPrefix(r.Children, "{")
+		r.Children = strings.TrimRight(r.Children, "}")
+		children := strings.Split(r.Children, ",")
+		childMap[r.IterationID] = children
+	}
 	countsMap := map[string]WICountsPerIteration{}
-	for _, iterationWithCount := range res {
-		countsMap[iterationWithCount.IterationID] = iterationWithCount
+	for _, i := range wiMap {
+		t := i.Total
+		c := i.Closed
+		if children, exists := childMap[i.IterationID]; exists {
+			for _, child := range children {
+				if _, exists := wiMap[child]; exists {
+					t += wiMap[child].Total
+					c += wiMap[child].Closed
+				}
+			}
+		}
+		countsMap[i.IterationID] = WICountsPerIteration{
+			IterationID: i.IterationID,
+			Total:       t,
+			Closed:      c,
+		}
 	}
 	return countsMap, nil
 }
 
 // GetCountsForIteration returns Closed and Total counts of WI for given iteration
 // It executes
 // SELECT count(*) as Total, count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed FROM "work_items" where fields@> concat('{"system.iteration": "%s"}')::jsonb and work_items.deleted_at is null
```

Comment:
You didn't updated the SQL comment when you modified the query. Or was the comment to the query further down the function? Please check the other comments as well.